### ISSUE TITLE
cp: copying dangling or nested symlink results in error

### ISSUE THREAD

**Issue by SuperSandro2000** [uutils/coreutils#2422]
I have the following directory structure where the file two points to a none existing file.

```
 tree -h a/
 a
└──  c
   └──  d
      └──  2 ⇒ ../../d/c/1
```

When copying with uutils-cp I get

```ShellSession
$ result/bin/cp -r -- a 2
cp: No such file or directory (os error 2)
```

Using gnu-cp works just fine and recreates the dangling symlink.

**Comment by SuperSandro2000** [comment]
Also experience the same error when copying this directory from the nix store

```ShellSession
result/bin/cp -r -- /nix/store/ayddsfvxqa8khxg5f0n5ysbml69m5cvf-source source
cp: '/nix/store/ayddsfvxqa8khxg5f0n5ysbml69m5cvf-source/misc/minetest-icon.ico' -> 'source/misc/minetest-icon.ico': No such file or directory (os error 2)
```

[source.tar.gz](https://github.com/uutils/coreutils/files/6666569/source.tar.gz)

**Comment by miDeb** [comment]
This should also be fixed by #2426 (I checked the example https://github.com/uutils/coreutils/issues/2422#issuecomment-862827133)
